ConcordiumWallet/AppCoordinator.swift

```diff
@@ -62,6 +62,27 @@ class AppCoordinator: NSObject, Coordinator, ShowAlert, RequestPasswordDelegate
         loginCoordinator.start()
     }
 
+    func openWalletConnectAccountSelection(url: URL) {
+        guard let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems else {
+            showErrorAlert(ViewError.simpleError(localizedReason: "WalletConnect URL format incorrect."))
+            return
+        }
+
+        if let uriValue = queryItems.first(where: { $0.name == "uri" })?.value {
+            if let coordinator = childCoordinators.first(where: { $0 is WalletConnectCoordinator }) as? WalletConnectCoordinator {
+                coordinator.start(with: uriValue)
+            } else {
+                let walletConnectCoordinator = WalletConnectCoordinator(
+                    navigationController: navigationController,
+                    dependencyProvider: defaultProvider,
+                    parentCoordinator: self
+                )
+                walletConnectCoordinator.start(with: uriValue)
+                childCoordinators.append(walletConnectCoordinator)
+            }
+        }
+    }
```

Comment:
Maybe this is a bit more explicit?

```suggestion
    func openWalletConnectAccountSelection(url: URL) {
        guard let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems else {
            showErrorAlert(ViewError.simpleError(localizedReason: "WalletConnect URL format incorrect."))
            return
        }

        if let uriValue = queryItems.first(where: { $0.name == "uri" })?.value {
            getOrCreateWalletConnectCoordinator().start(with: uriValue)
        }
    }
    
    func getOrCreateWalletConnectCoordinator() -> WalletConnectCoordinator {
        if let c = childCoordinators.first(where: { $0 is WalletConnectCoordinator }) as? WalletConnectCoordinator {
            return c
        }
        let c = WalletConnectCoordinator(
            navigationController: navigationController,
            dependencyProvider: defaultProvider,
            parentCoordinator: self
        )
        childCoordinators.append(c)
        return c
    }
```

Reply:
But what if the user is already in the WC flow? Wouldn't it be better to close that flow first then? I.e. using `dismissWalletConnectCoordinator`.

Reply:
hmm that would be the best approach correct

Reply:
In fact, if there is already a WC connection, we still get the duplicate account list. I then got into a state where the disconnect button no longer worked.

I think connecting via deep link while already connected might be enough of an edge case that we shouldn't worry about it, but it's worth checking if the other solution avoids the issue.